xio-core/src/main/java/com/xjeffrose/xio/http/Client.java

```diff
@@ -56,13 +59,33 @@ public ChannelFuture write(Request request) {
       ChannelFuture future = connect();
       channel = future.channel();
       ChannelPromise promise = channel.newPromise();
-      PromiseCombiner combiner = new PromiseCombiner();
-      combiner.add(future.addListener(connectionListener));
-      combiner.add(channel.writeAndFlush(request).addListener(writeListener));
-      combiner.finish(promise);
+      future.addListeners(connectionListener, new GenericFutureListener<Future<? super Void>>() {
```

Comment:
a lambda would be preferred rather than an anonymous GenericFutureListener

Reply:
too bad `ChannelFuture` does not have a monadic interface - a flatMap or andThen would make this easier to read